lib/MusicBrainz/Server/Validation.pm

```diff
@@ -102,6 +103,13 @@ sub is_database_bigint_id {
     is_positive_integer($t) and $t <= $MAX_POSTGRES_BIGINT;
 }
 
+sub is_database_indexable_string {
```

Comment:
Why do we need both of this and the 1024 limit? Wouldn't this check be enough for our needs? Or is it just to have a more easily communicable limit?

Reply:
> I did check those, but I'm still a bit confused. It seems the real limit that triggered the bug was actually the 2704 bytes one, no?

The bug MBS-13555, yes. And it would also prevent MBS-13536 to happen again.

> Wouldn't it be enough then to check for 2704 bytes and not worry about the amount of characters and whether they are multibyte or not?

As explained in commit message, it is more reasonable to additionally limit to 1024 characters.

Reply:
IIUC, the 8192 limit from MBS-13536 is for all indexes generally, and the 2704 limit is for btree indexes. The index from MBS-13536 was *also* a btree index, but perhaps it hit the "general" limit first before it checked the more specifiec btree limits (but I am just guessing).

> As explained in commit message, it is more reasonable to additionally limit to 1024 characters.

I agree that a fixed character limit is reasonable and also easier to explain to users -- though I'm not sure if the majority of the offenders are mistakes or intended to cause harm:

1055 characters - https://musicbrainz.org/edit/111225345
1056 characters - https://musicbrainz.org/edit/111225330
1126 characters - https://musicbrainz.org/edit/111225347
1176 characters - https://musicbrainz.org/edit/111225344
1377 characters - https://musicbrainz.org/edit/111225342

I'm not familiar with any of the source material though.

Given the additional btree-indexable byte limit check, we could perhaps be a little more generous and use a less-geeky number like 1500 which would cover most of the existing cases. Or maybe allow auto-editors to bypass the limit (as long as it's under 2704 bytes still).

Reply:
These are all examples of mistakes:

> 1055 characters - https://musicbrainz.org/edit/111225345

This is a clearly list of works that should have been written in the liner notes and would better fit in work relationships, or in annotation to start with.

It should be: “Musicalische Exequien, SWV 279, Concert in Form einer teutschen Begräbnis-Missa: II”

> 1056 characters - https://musicbrainz.org/edit/111225330

Same: “Transmission 7”

> 1126 characters - https://musicbrainz.org/edit/111225347

Same: “24 Duos Op.186 (For Two Saxophones)”

> 1176 characters - https://musicbrainz.org/edit/111225344

Same: “Korean Cosmetics”

No work relationship here though. This is actually a recipe, it would just better fit in annotation. The above name of the recipe is even missing from the current data if we compare to the linked [Bandcamp page](https://absenterratum.bandcamp.com/track/ingredients-water-eau-methylpropanediol-glycerin-ethyl-perfluoroisobutyl-ether-ethyl-perfluorobutyl-ether-sapindus-mukorossi-fruit-extract-acrylates-copolymer-betaine-hydroxyacetophenone-polyacry). Just using this name as track title would make the tracklist much more readable.

> 1377 characters - https://musicbrainz.org/edit/111225342

Same: “New Numbers”

Reply:
Thanks for checking them; it'd be nice to replace your existing edits with the suggestions you're certain about when you have time, since it's much nicer than truncation, and it may be unlikely for some of these releases to be fixed for a long time! (Note I didn't link to any of the "creative writing" ones.)

Reply:
The 24 Duos one is a wrong Discogs import where a lot of tracks got smashed together, I'll fix that one. I disagree that Transmission 7 is wrong, but I think it'd be ok either way - if you change that one you should change all though. "Korean Cosmetics" seems to be the artist name and should probably not be added to the title - I'd just find a good place to put `...` at the end of that one, maybe just do `Moringa Pterygosperma Seed Extract...`? New Numbers seems fine to change to just that. The Musicalische Exequien, SWV 279 is also not wrong, but shortening it is probably sensible. That disc has other errors, so I'll do that part myself as well.

Reply:
Sorry but no, the changes should probably be made for the whole releases not just these tracks, this isn’t worth my time, and there is a lot of trash stuff around (such as the above second example with 1056 characters) that I don’t even want to read, that’s why I made a script to truncate names arbitrarily rather than editing by hand. I’m not a classical editor either, so the classical releases would be better edited by some more knowledgeable editor than me about classical releases. I’m sure that some interested editors are following these edits that are linked from the [ticket’s comment](https://tickets.metabrainz.org/browse/MBS-13562?focusedId=72046&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-72046) and will make the most appropriate changes in time (likely after this check is deployed). As I wrote on IRC before, the data isn’t completely lost since it lies in the editing history anyway. There should be nothing wrong with truncating a few names arbitrarily for the sake of maintaining the database. If we decide to relax this constraint later on, it will just take changing a constant in code and reverting a few edits.

Reply:
Thanks @reosarevok -- I guess we're all okay with the existing checks then. I didn't mean to start bikeshedding on a specific number, but was just hesitant to restrict possibly-valid data. @yvanzo, it sounds like you can just cancel your existing edits if reosarevok is looking into them.

Reply:
> Thanks @reosarevok -- I guess we're all okay with the existing checks then. I didn't mean to start bikeshedding on a specific number, but was just hesitant to restrict possibly-valid data.

Initially I was more leaning towards 512 bytes to be honest. I picked up 1024 bytes after your own data analysis on IRC as there is indeed no legitimate use case beyond it. Just reminding that [titles](https://en.wikipedia.org/wiki/Title_(publishing)) are supposed to be “minimal summaries”, neither dissertations nor liner notes.

> @yvanzo, it sounds like you can just cancel your existing edits if reosarevok is looking into them.

It wouldn’t hurt to approve all of these edits really, even if these titles are further trimmed later on by volunteers.